substratest/assets.py

```diff
@@ -168,15 +194,23 @@ def load(cls, d):
             raise errors.TError(f"cannot parse asset `{d}`") from e
 
 
-class _Asset(_DataclassLoader, abc.ABC):
+class _InternalStruct(pydantic.BaseModel, _DataclassLoader, abc.ABC):
+    """Internal nested structure"""
+
+
+class _Asset(_InternalStruct, abc.ABC):
     """Represents assets stored in the Substra framework.
 
     Convert a dict with camel case fields to a Dataclass.
     """
 
 
-@dataclasses.dataclass(frozen=True)
-class Permission(_DataclassLoader):
+class _Frozen:
+    class Config:
+        allow_mutation = False
```

Comment:
It looks like this is not working as expected. I tried the following:

```python 
In [3]: n = Node(id='foo', is_current=False)

In [4]: n
Out[4]: Node(id='foo', is_current=False)

In [5]: n.id = 'bar'

In [6]: n.id
Out[6]: 'bar'
```

It should instead have raised `TypeError: "Node" is immutable and does not support item assignment`

Reply:
Yes indeed. I'll look for a solution 👍

Reply:
I'm not sure this constraint is bringing value to the tests library.
Even if the goal of this PR is to keep the same perimeter, in my opinion, we can remove the immutable assets feature.